providers/prefab/pkg/provider_test.go

```diff
@@ -363,7 +365,12 @@ func TestMain(m *testing.M) {
 		os.Exit(1)
 	}
 
-	of.SetProviderAndWait(provider)
+	err = of.SetProviderAndWait(provider)
+	if err != nil {
+		fmt.Printf("Error during provider set: %v\n", err)
+		os.Exit(1)
```

Comment:
Do we really want to crash the test with `os.Exit` ?
It will probably better to return assert here rather than crashing the full test suite.

Reply:
This is `TestMain(m *testing.M)` and from [docs](https://pkg.go.dev/testing#hdr-Main)

>> `m.Run` will return an exit code that may be passed to [os.Exit](https://pkg.go.dev/os#Exit)